### PR TITLE
Better intellij dev experience

### PR DESCRIPTION
**Pull request by edgao**
I'm sure there's more cleanup that could happen in the plugin file. But presumably we want to just nuke the entire airbyteDocker plugin eventually, so not going to bother.

airbyteDocker now explicitly declares `build/distributions` as an input directory, which I believe is what most (all?) java connectors copy into their docker image. This probably misses a few edge cases, but given that this only matters for running locally in intelllij - I think that's OK.

(confirmed locally that running tests twice in a row skips airbyteDocker correctly)

### CODE REVIEW

**Comment by evantahler** [comment]
Tagging @alafanechere - this completes the circle 🟢!

**Comment by evantahler** [comment]
@edgao so like... does it work?  Can we get a loom of you using it?
* run `docker system prune -a`
* run a test in inteliJ (time it)
* run that test in inteliJ a second time (is it faster?)

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`4cae645174`](https://github.com/airbytehq/airbyte/pull/28817/commits/4cae645174828d6eccb7b431174cb878ff6782ff)) - ✅

⏲️  Total pipeline duration: 63mn12s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1690499566/4cae645174828d6eccb7b431174cb878ff6782ff/destination-snowflake/1.2.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by edgao** [comment]
https://www.loom.com/share/69e8677d903d48d6821939b2e1cd96ea

it does run successfully, but I think airbyte-ci isn't actually caching successfully on my laptop. Ran it twice in succession and it took >9 minutes both times (nearly 10 minutes on the first run).

the old airbyteDocker was significantly faster, but I think that's a bit misleading - it also wasn't reliably building new images, so I was manually running airbyte-ci before tests anyway. I.e. I think this is still an improvement for local dev.

I'll try and find a way to disable this on CI for now. The CI test run on this branch took about 10 minutes longer to run `integrationTest` (54mn39s vs [44mn04s](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/nightly_builds/master/1690762548/ad12b1e1b20713482ec2e6c3b6e8ae95b86e6dbd/destination-snowflake/1.2.4/output.html) on the latest nightly)... which might indicate more general caching issues, actually

**Comment by edgao** [comment]
also, if you want to try this out locally - if you haven't rebooted since switching to the poetry-based airbyte-ci install:
* restart intellij
* `kill` all your gradle daemon processes

otherwise you might get weird issues where they can't find airbyte-ci

**Review by jbfbell — APPROVED** [review]

File: buildSrc/src/main/groovy/airbyte-docker.gradle [hunk 1]
```diff
@@ -194,7 +194,13 @@ class AirbyteDockerPlugin implements Plugin<Project> {
     }
 
     static def createTask(Project project, String taskName, String dockerFile) {
-        if (project.file(dockerFile).exists()) {
+        // airbyte-ci only supports actual connectors, so exclude all the base modules
+        if (project.name.startsWith('source-') || project.name.startsWith('destination-')) {
+            project.task(taskName, type: Exec) {
+                workingDir project.rootProject.projectDir
+                commandLine "airbyte-ci", "connectors", "--name", project.name, "build"
```
Comment (alafanechere) [inline buildSrc/src/main/groovy/airbyte-docker.gradle:201]: The current `airbyte-ci` call starts a terminal UI which won't play very well with execution from gradle
```suggestion
                commandLine "airbyte-ci", "--no-tui", "connectors", "--name", project.name, "build"
```

**Review by alafanechere — COMMENTED** [review]
This is great and a first nice step in sunsetting `airbyteDocker` right? 
Please note my comment; LGTM otherwise.
Would we want to handle the `airbyte-ci` installation from `gradle`? (might be overkill ATM for local gradle execution)

**Comment by alafanechere** [comment]
> I'll try and find a way to disable this on CI for now. The CI test run on this branch took about 10 minutes longer to run integrationTest (54mn39s vs [44mn04s](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/nightly_builds/master/1690762548/ad12b1e1b20713482ec2e6c3b6e8ae95b86e6dbd/destination-snowflake/1.2.4/output.html) on the latest nightly)... which might indicate more general caching issues, actually

When `airbyte-ci` runs integration test or `build` it's using a gradle command with `-x airbyteDocker` : I think it means that on the CI `airbyte-ci` won't call itself.

**Comment by evantahler** [comment]
@edgao I think we can keep this task fast when nothing has changed if we modify the `isUpToDate()` method.  I think this method is expecting `project.rootProject.imageToHash.get` to have been set by the build command, which are no longer doing.  We could set that after shelling out to `airbyte-ci`, or be lazier and skip the hash check, and just be OK if the image exists at all

**Comment by edgao** [comment]
I dropped the uptodate check entirely b/c I thought airbyte-ci would handle it 🤦

I never quite figured out what the imageToHash thing actually does, and suspect it over and/or undercaches things... maybe gradle has something for "run this task only if compileJava did anything"? Which probably works >95% of the time (i.e. it misses changes to shell scripts/etc, but we basically never change those files anyway, and we can always come back to this))

**Comment by edgao** [comment]
airbyteDocker now explicitly declares `build/distributions` as an input directory, which I believe is what most (all?)) java connectors copy into their docker image. This probably misses a few edge cases, but given that this only matters for running locally in intelllij - I think that's OK. PTAL!

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`ab57ea5aa6`](https://github.com/airbytehq/airbyte/pull/28817/commits/ab57ea5aa60fbe921ef2512015d119b8cf11c8dd)) - ❌

⏲️  Total pipeline duration: 41mn49s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691711964/ab57ea5aa60fbe921ef2512015d119b8cf11c8dd/destination-snowflake/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ab57ea5aa60fbe921ef2512015d119b8cf11c8dd)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`8442d5ea38`](https://github.com/airbytehq/airbyte/pull/28817/commits/8442d5ea38f08ae0d57967474418837422a1d9e5)) - ❌

⏲️  Total pipeline duration: 40mn40s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691714600/8442d5ea38f08ae0d57967474418837422a1d9e5/destination-snowflake/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8442d5ea38f08ae0d57967474418837422a1d9e5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by alafanechere** [comment]
@edgao you're right considering the airbyte-ci gradle caching does not properly work.
I logged this issue to improve it https://github.com/airbytehq/airbyte/issues/28983
Conor made it work in `aircmd` (the `with_gradle` function).

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`8cf67b4a82`](https://github.com/airbytehq/airbyte/pull/28817/commits/8cf67b4a82695638ed0b62b3e18a9fed69894347)) - ❌

⏲️  Total pipeline duration: 04mn19s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691774874/8cf67b4a82695638ed0b62b3e18a9fed69894347/destination-snowflake/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8cf67b4a82695638ed0b62b3e18a9fed69894347)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`8cf67b4a82`](https://github.com/airbytehq/airbyte/pull/28817/commits/8cf67b4a82695638ed0b62b3e18a9fed69894347)) - ❌

⏲️  Total pipeline duration: 33mn18s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691774874/8cf67b4a82695638ed0b62b3e18a9fed69894347/destination-bigquery/1.7.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8cf67b4a82695638ed0b62b3e18a9fed69894347)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by edgao** [comment]
connectors base build failed on CI:
```
2023-08-11T17:59:36.0531571Z 1: Task failed with an exception.
2023-08-11T17:59:36.0532505Z -----------
2023-08-11T17:59:36.0532899Z * What went wrong:
2023-08-11T17:59:36.0533604Z Execution failed for task ':airbyte-integrations:connectors:destination-bigquery:airbyteDocker'.
2023-08-11T17:59:36.0534290Z > A problem occurred starting process 'command 'airbyte-ci''
```

but no other information :( will keep poking at this

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`e1b4fdc8da`](https://github.com/airbytehq/airbyte/pull/28817/commits/e1b4fdc8da2d6df2ab56b9debf85177f518e613b)) - ❌

⏲️  Total pipeline duration: 42mn03s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691788214/e1b4fdc8da2d6df2ab56b9debf85177f518e613b/destination-snowflake/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e1b4fdc8da2d6df2ab56b9debf85177f518e613b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`e1b4fdc8da`](https://github.com/airbytehq/airbyte/pull/28817/commits/e1b4fdc8da2d6df2ab56b9debf85177f518e613b)) - ❌

⏲️  Total pipeline duration: 30mn44s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691788214/e1b4fdc8da2d6df2ab56b9debf85177f518e613b/destination-bigquery/1.7.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e1b4fdc8da2d6df2ab56b9debf85177f518e613b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by edgao** [comment]
uhhhh I'm going to just remove the connectors from the connectors_base sub build. settings.gradle claims they're only needed for normalization integration tests, which we're not running anymore.

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`1cde8d0dee`](https://github.com/airbytehq/airbyte/pull/28817/commits/1cde8d0dee994a2ba75fddddc170788c8c2e2599)) - ❌

⏲️  Total pipeline duration: 31mn36s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691796370/1cde8d0dee994a2ba75fddddc170788c8c2e2599/destination-bigquery/1.7.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1cde8d0dee994a2ba75fddddc170788c8c2e2599)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`1cde8d0dee`](https://github.com/airbytehq/airbyte/pull/28817/commits/1cde8d0dee994a2ba75fddddc170788c8c2e2599)) - ❌

⏲️  Total pipeline duration: 40mn01s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_airbytedocker_airbyte_ci/1691796370/1cde8d0dee994a2ba75fddddc170788c8c2e2599/destination-snowflake/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1cde8d0dee994a2ba75fddddc170788c8c2e2599)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Pull request closed without merging**